bin/nyc.js

```diff
@@ -36,21 +34,60 @@ if ([
   }
   if (argv.all) nyc.addAllFiles()
 
-  var env = {
-    // Support running nyc as a user without HOME (e.g. linux 'nobody'),
-    // https://github.com/istanbuljs/nyc/issues/951
-    SPAWN_WRAP_SHIM_ROOT: process.env.SPAWN_WRAP_SHIM_ROOT || process.env.XDG_CACHE_HOME || require('os').homedir(),
-    NYC_CONFIG: JSON.stringify(argv),
-    NYC_CWD: process.cwd()
-  }
+  if (argv.setNodeOptions) {
+    const wrapper = require.resolve('../lib/wrap.js')
+    let nodeRequire
+    /*
+     * Node.js 11 and older do not support specifying `--require "/path/with spaces"`.
+     *  - For node.js 12+ we use the proper quoted syntax.
+     *  - For older versions we just directly use the path to lib/wrap.js as long as
+     *    it does not contain any spaces.
+     *  - For older versions when the path to lib/wrap.js contains a space we must
+     *    combine NODE_PATH and NODE_OPTIONS.
+     *
+     * Ref https://github.com/nodejs/node/pull/24065
+     */
+    /* istanbul ignore next */
```

Comment:
The branches following this line are version dependent.  I have tested all three branches locally using node.js 8, 10 and 12.  I've tested with nyc in a directory with and without spaces and double-quote.

Reply:
Does the ENOMEM error still occur on Windows with node 10+?

Reply:
It does still occur, specifically on 10.16.0 and above.  Technically we could add `10.15.0` into testing.

See test log from yesterday when testing node-preload - https://travis-ci.org/cfware/node-preload/builds/582661358